src/util/codec/bytes.rs

```diff
@@ -112,10 +113,19 @@ pub fn encoded_compact_len(mut encoded: &[u8]) -> usize {
     vn + (encoded.as_ptr() as usize - last_encoded)
 }
 
-pub trait CompactBytesFromFileDecoder: NumberDecoder {
+pub trait CompactBytesFromFileDecoder: Read {
```

Comment:
Use `BufRead` instead.

Reply:
Seems can't since the trait `std::io::BufRead` is not implemented for `std::fs::File`